Text/LaTeX/Base/Syntax.hs

```diff
@@ -180,7 +180,7 @@ matchCommand :: (String -> Bool) -> LaTeX -> [(String,[TeXArg])]
 matchCommand f (TeXComm str as) =
   let xs = concatMap (matchCommandArg f) as
   in  if f str then (str,as) : xs else xs
-matchCommand f (TeXCommS str) = if f str then [(str,[])] else []
+matchCommand f (TeXCommS str) = [(str, []) | f str]
```

Comment:
I'm ok with the list comprehension, but it's debatable if it's really better style.

Reply:
I agree that it is not per se better. The arguments pro are probably that it is more compact, and is more self-explaining. But I agree that `hlint` perhaps should not give this as a hint, since it is not per se better than using the `if ... then ... else` expression.

Reply:
Sorry to be late here. I don't really mind this change, it's fine, but I don't see how this is clearer than the original code.